sys/include/entropy_source/entropy_source.h

```diff
@@ -0,0 +1,295 @@
+
+    state->old_sample = 0;
+    state->cnt_rep = 0;
+    state->c_rep = c_rep;
+}
+
+/**
+ * @brief   Initialize structure for Adaptive Proportion Test
+ *
+ * @param[in, out] state   Test structure of one entropy source.
+ * @param[in]      c_prop  Cutoff value calculated by @ref entropy_source_test_prop_cutoff.
+ */
+static inline void entropy_source_test_prop_init(
+    entropy_source_tests_prop_t *state, uint16_t c_prop)
+{
+    assert(state != NULL);
+
+    state->old_sample = 0;
+    state->cnt_prop = 0;
+    state->cnt_window = CONFIG_ENTROPY_SOURCE_TESTS_WIN;
+    state->c_prop = c_prop;
+}
+
+/**
+ * @brief   Performs Repetition Count Test (NIST SP 800-90B 4.4.1).
+ *
+ * This function will not block sampling. It only indicates detected errors.
+ *
+ * @param[in, out] state    Test structure of one entropy source.
+ * @param[in]      sample   Current sample.
+ *
+ * @return          ENTROPY_SOURCE_OK on success
+ * @return          ENTROPY_SOURCE_ERR_TEST_REP on detected weakness
+ */
+int entropy_source_test_rep(entropy_source_tests_rep_t *state, uint8_t sample);
+
+/**
+ * @brief   Performs Adaptive Proportion Test (NIST SP 800-90B 4.4.2).
+ *
+ * This function will not block the sampling. It only indicates detected errors.
+ *
+ * @param[in, out] state   Test structure of one entropy source.
+ * @param[in]      sample  current sample.
+ *
+ * @return          ENTROPY_SOURCE_OK on success
+ * @return          ENTROPY_SOURCE_ERR_TEST_PROP on detected weakness
+ */
+int entropy_source_test_prop(entropy_source_tests_prop_t *state,
+                             uint8_t sample);
+
+
+/**
+ * @brief   Convenience function to perform @ref entropy_source_test_rep
+ *          and @ref entropy_source_test_prop.
+ *
+ * This function will not block the sampling. It only indicates detected errors.
+ *
+ * @param[in, out] state_rep    Repetition Count test structure of one entropy source.
+ * @param[in, out] state_prop   Adaptive Proportion test structure of one entropy source.
+ * @param[in]      sample       Current sample.
+ *
+ * @return          ENTROPY_SOURCE_OK on success
+ * @return          negative @ref entropy_source_error_t code on error
+ */
+static inline int entropy_source_test(entropy_source_tests_rep_t *state_rep,
+                                      entropy_source_tests_prop_t *state_prop,
+                                      uint8_t sample)
+{
+    int ret = ENTROPY_SOURCE_OK;
+
+    if (entropy_source_test_rep(state_rep, sample) < 0) {
+        ret = ENTROPY_SOURCE_ERR_TEST_REP;
+    }
+    if (entropy_source_test_prop(state_prop, sample) < 0) {
+        /* If repetition count failed before, indicate that both tests failed */
+        if (ret == ENTROPY_SOURCE_ERR_TEST_REP) {
+            ret = ENTROPY_SOURCE_ERR_TEST_BOTH;
+        }
+        else {
+            ret = ENTROPY_SOURCE_ERR_TEST_PROP;
+        }
+    }
+    return ret;
+}
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* ENTROPY_SOURCE_H */
```

Comment:
```suggestion
#endif /* ENTROPY_SOURCE_ENTROPY_SOURCE_H */
```

if you don't want to move this file.